### ISSUE TITLE
undefined Output should show error in the parent template

### ISSUE THREAD

**Issue by urielzen** [angular/angular#43603]
### Which @angular/* package(s) are relevant/releated to the feature request?

core

### Description

I believe I have all the template strict checks enabled, and I was expecting to have a template error at my parent component telling me that my child component does not have certain output if there is a typo in its name.

```typescript
@Component({
  selector: 'app-child',
  templateUrl: './child.component.html'
})
export class ChildComponent {
  @Output() outputFromChild = new EventEmitter<string>();
}
```

```html
<app-child (outputFromChildWithATypo)="handleOutputAtParent($event)"></app-child>
```

### Proposed solution

The compiler should warn about a non-existing output in the component element.

### Alternatives considered

Please let me know if this error check already exists, I may just be missing enabling it. I tried googling for it but could not find anything relevant.

**Comment by JoostK** [comment]
There is indeed a gap in type-checking here. The misnamed output does have a valid target though: the `app-child` DOM element that is rendered. We don't currently check output against the DOM schema, so any unknown output is not reported.

You can partially workaround [using `"strictDomEventTypes": true`](https://angular.io/guide/template-typecheck), as that would have made `$event` of the DOM's native `Event` type.

**Comment by alxhub** [comment]
As @JoostK says, the issue here is really that binding to a DOM event is completely legal in Angular, even when the element is otherwise a component. However, there's definitely some opportunity here. We could make it a _warning_ to do this (in fact, we could make `strictDomEventTypes` present as a warning by default as well).

@dgp1130, another one for your list!

**Comment by dgp1130** [comment]
I agree this is a good candidate for an extended diagnostic. It's really the split-two-way-binding error, just only one-way.

Definitely if you're adding the output binding to an Angular component, it should be a an `@Output` field. Maybe native events could be allowed (`click`), but I'm not sure if we support components being used in that way (ie. with `role="button"`).

**Comment by nbnds** [comment]
This issue has also an effect described in https://github.com/angular/angular/issues/45996

**Comment by ippeiukai** [comment]
Just an idea:

Like attributes (`[attr.foo]`) vs input properties (`[foo]`) , we could
- bind `(event.click)` to DOM events
- bind `(click)` to 'click' output property only (in future, but only deprecate DOM event usage for a while)
- add an angularCompilerOption to strictly detect unknown output property (opt-in for existing projects for a while)

**Comment by mc-paula** [comment]
@JoostK Would it be possible to reopen the voting? I think this feature would benefit many developers. Thank you :)

**Comment by JoostK** [comment]
No need, since this is labeled "Under consideration"

**Comment by nickbullock** [comment]
Hey @JoostK @alxhub 

We do have `strictTemplates: true`, so `strictDomEventTypes` is also turned on, but just faced the same issue with not existing output not being reported

```
Angular CLI: 16.2.4
Node: 16.17.1
Package Manager: npm 8.15.0
OS: darwin arm64

Angular: 16.2.7
... animations, common, compiler, compiler-cli, core, elements
... forms, platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1602.4
@angular-devkit/build-angular   16.2.4
@angular-devkit/core            16.2.4
@angular-devkit/schematics      16.2.4
@angular/cdk                    16.2.6
@angular/cli                    16.2.4
@schematics/angular             16.2.4
rxjs                            7.8.1
typescript                      5.1.6
webpack                         5.88.2
zone.js                         0.13.1
```

**Comment by JoostK** [comment]
> We do have `strictTemplates: true`, so `strictDomEventTypes` is also turned on

These options aren't enabled together, `strictDomEventTypes` has to be enabled by itself on top of `strictTemplates`.

**Comment by nickbullock** [comment]
thanks @JoostK! 
added it on top of `strictTemplates` but the result is same - no errors

**Comment by siegenthalerroger** [comment]
@JoostK if they aren't enabled together then the documentation is misleading (and missing that this is the case): https://angular.dev/tools/cli/template-typecheck#troubleshooting-template-errors:~:text=Unless%20otherwise%20commented,other%20way%20around).

Looking at the code it does seem like they're both activated together: https://github.com/angular/angular/blob/7eb9fcd6a07e39eb8fe43ef0576c9172bd4b59e2/packages/compiler-cli/src/ngtsc/core/src/compiler.ts#L1077

_I didn't see any commits that changed this in the last year, so I thought this worth mentioning_

**Comment by JeanMeche** [comment]
Just to clarify: setting `"strictDomEventTypes": true` does not flag unknown outputs, it will report the right type of `$event`. This is often enough to raise an error. 

Also `strictDomEventTypes` is set with `strictTemplates` but isn't with the deprecated `fullTemplateCheck`.

**Comment by JeanMeche** [comment]
We've landed #70527 in 22.2 to improve the situation. 

The new check is behind a flag for now to be non-breaking: Set `strictUnclaimedEventNames: true` to have your event bindings checked (little caveat is that lower cases events will still be skiped, at least for now)
